packages/parameters/src/appconfig/AppConfigProvider.ts

```diff
@@ -13,6 +13,7 @@ import type {
 class AppConfigProvider extends BaseProvider {
   public client: AppConfigDataClient;
   protected configurationTokenStore: Map<string, string> = new Map();
+  protected valueStore: Map<string, Uint8Array> = new Map();
```

Comment:
I don't think this change is necessary. 

The `AppConfigProvider` extends the `BaseProvider` which is the one that holds the caching logic.

On a high level, the call stack goes like this:
- User runs `provider.get('...');`
- `AppConfigProvider.get()` runs first, and calls `BaseProvider.get()` passing down the same args ([here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/parameters/src/appconfig/AppConfigProvider.ts#L53))
- Within `BaseProvider.get()`
  - The first action is to check if the value is in the cache & it's not expired, if both are `true`, the value is returned from cache ([here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/parameters/src/BaseProvider.ts#L47-L51))
  - If there's no cached value, then `AppConfigProvider._get` (this method) is called, and we hit the API ([here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/parameters/src/BaseProvider.ts#L55))
  - After that, we apply transformations on the returned value (if any), and we add it to cache

With this in mind, I don't think we should add a second cache here. We can consider improving the documentation around this, but if users expect to request a config very often they should handle this with the original `maxAge` setting. This will ensure that the value is returned from cache and won't be empty. Users can also influence the refresh rate from AppConfig by passing a parameter to the API call via `sdkOptions` to refresh more often.
But I don't think we should cache indefinitely and again here.

Also note that this is the same behavior/implementation found in the Python version of this library.

Reply:
I replied to this thread down in a separate comment. I suppose it comes down to how much provider implementation detail we decide to expose to clients.

With that said, I'm _fairly_ sure the Python library is using an intermediate cache (unless I'm misreading!?):
https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/aws_lambda_powertools/utilities/parameters/appconfig.py#L121

Reply:
Either way, if you'd prefer that the `AppConfig` behavior be exposed, this PR can probably be closed. I'll re-test the original `maxAge` algorithm and refocus only there.

Reply:
No, you were right. Let's continue with this change, see this comment for full comment.

Marking this as resolved.